spring-cloud-app-broker-security-credhub/src/main/java/org/springframework/cloud/appbroker/workflow/binding/CredHubPersistingDeleteServiceInstanceBindingWorkflow.java

```diff
@@ -46,18 +50,43 @@ public CredHubPersistingDeleteServiceInstanceBindingWorkflow(ReactiveCredHubOper
 		DeleteServiceInstanceBindingResponseBuilder responseBuilder) {
 		return buildCredentialName(request.getServiceDefinitionId(), request.getBindingId())
 			.filterWhen(this::credentialExists)
-			.flatMap(credentialName -> credHubOperations.credentials()
-				.deleteByName(credentialName)
-				.doOnRequest(
-					l -> LOG.debug("Deleting binding credentials with name '{}' in CredHub", credentialName.getName()))
-				.doOnSuccess(r -> LOG
-					.debug("Finished deleting binding credentials with name '{}' in CredHub", credentialName.getName()))
-				.doOnError(exception -> LOG.error(
-					String.format("Error deleting binding credentials with name '%s' in CredHub with error: '%s'",
-						credentialName.getName(), exception.getMessage()), exception)))
+			.flatMap(deleteCredential())
+			.flatMap(deletePermission())
 			.thenReturn(responseBuilder);
 	}
 
+	private Function<ServiceInstanceCredentialName, Mono<ServiceInstanceCredentialName>> deleteCredential() {
```

Comment:
My preference is to return Reactor types instead of functions. I find it more readable and maintainable. That said, this is perfectly fine and looks clean. Happy to discuss further.